qemu/tests/virtio_mem_reboot.py

```diff
@@ -0,0 +1,62 @@
+import time
+
+from virttest import error_context
+
+from virttest.utils_misc import normalize_data_size
+from provider import virtio_mem_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Virtio-mem reboot test
+    1) Boot guest
+    2) Check virtio-mem device
+    4) Resize virtio-mem device
+    5) Check virtio-mem device
+    6) Resize virtio-mem device
+    7) Check virtio-mem device
+    8) Reboot guest
+    9) Check virtio-mem device
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    threshold = params.get_numeric("threshold", target_type=float)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    vm.wait_for_login()
+
+    virtio_mem_model = 'virtio-mem-pci'
+    if '-mmio:' in params.get("machine_type"):
+        virtio_mem_model = 'virtio-mem-device'
+    vmem_dev = vm.devices.get_by_params({'driver': virtio_mem_model})[0]
+    device_id = vmem_dev.get_qid()
+    requested_size_vmem_test = params.get("requested-size_test_vmem0")
+    node_id = int(vmem_dev.get_param('node'))
+    req_size = vmem_dev.get_param('requested-size')
+    initial_req_size = str(int(float(normalize_data_size(req_size, 'B'))))
+
+    virtio_mem_utils.check_memory_devices(device_id, initial_req_size,
+                                          threshold, vm, test)
+    virtio_mem_utils.check_numa_plugged_mem(node_id, initial_req_size,
+                                            threshold, vm, test)
+    for requested_size in requested_size_vmem_test.split():
+        req_size_normalized = int(float(normalize_data_size(requested_size,
+                                  'B')))
+        vm.monitor.qom_set(device_id, "requested-size",
+                           req_size_normalized)
+        time.sleep(30)
+        virtio_mem_utils.check_memory_devices(device_id, requested_size,
+                                              threshold, vm, test)
+        virtio_mem_utils.check_numa_plugged_mem(node_id, requested_size,
+                                                threshold, vm, test)
+    vm.reboot()
```

Comment:
Why don't we just reuse `virtio_mem_numa_basic` plus a reboot step?

Reply:
@zhenyzha that's a good point, you mean to modify existing virtio_mem_numa_basic code and make it valid for both cases, right?

Reply:
Yes, I think this can also meet the testing needs. 
Of course, if you have other considerations, you can also list them :)

Reply:
@yanan-fu what do you think about mixing this two cases? I've given a try and seems it could be possible

Reply:
I have the same idea with @zhenyzha , the steps for these 2 cases are almost same. 
suggest to combine them into one script and cfg with different case variants.